torchgeo/datasets/glacier_calving_front.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Glacier Calving Front dataset."""
+
+import glob
+import os
+from collections.abc import Callable
+from typing import ClassVar
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_and_extract_archive, extract_archive
+
+
+class GlacierCalvingFront(NonGeoDataset):
+    """Glacier Calving Front dataset.
+
+    The `Glacier Calving Front <https://doi.pangaea.de/10.1594/PANGAEA.940950>`__ dataset is a
+    semantic segmentation dataset of marine-terminating glaciers.
+
+    Dataset features:
+
+    * 46,700 train, 18,744 validation, and 28,376 test images
+    * varying spatial resolution of 6-20m
+    * paired binary calving front segmentation masks
+    * paired multi-class land cover segmentation masks
+
+    Dataset format:
+
+    * images are single-channel pngs with dimension 256x256
+    * segmentation masks are single-channel pngs
+
+    Dataset classes:
+
+    1. background
+    2. ocean
+    3. rock
+    4. glacier
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://essd.copernicus.org/articles/14/4287/2022/
+
+    .. versionadded:: 0.7
+    """
+
+    valid_splits = ('train', 'val', 'test')
+
+    zipfilename = 'glacier_calving_data.zip'
+
+    data_dir = 'glacier_calving_data'
+
+    image_dir = 'sar_images'
+
+    mask_dirs = ('fronts', 'zones')
+
+    url = 'https://huggingface.co/datasets/torchgeo/glacier_calving_front/resolve/main/glacier_calving_data.zip'
+
+    md5 = '56e39e33f88a9842f48c513083e3c50a'
+
+    px_class_values: ClassVar[dict[int, str]] = {
+        0: 'background',
+        64: 'ocean',
+        127: 'rock',
+        254: 'glacier',
```

Comment:
Either in the dataset or the datamodule, we need to map these to ordinal numbers, correct? I've been meaning to add a transform for this since it comes up so often.

Reply:
I marked as a draft, because I am not 100% sure here, that's why I asked the question to the author. But yeas you are correct, missing code for mapping it to ordinal and will do that once I get the answer :)